### DISCUSSION TITLE
Editor Support

### DISCUSSION THREAD

**Discussion by MichaReiser** [rome/tools#3544] [Polls]
Poll related to #2390 to understand the priority of individual editors. 

Rome supports VS Code today.

If you're rooting for a JetBrains IDE, please upvote [the Rome issue](https://youtrack.jetbrains.com/issue/WEB-46895/Support-for-Romejs)

**Comment by ruijdacd** [comment]
[Zed](https://zed.dev/)

**Reply by keinsell** [reply]
As a temporary solution you can following to your zed config.

```
  "language_overrides": {
    "TypeScript": {
      "format_on_save": {
        "external": {
          "command": "node_modules/.bin/rome",
          "arguments": [
            "format",
            "--stdin-file-path",
            "{buffer_path}"
          ]
        }
      }
    }
  }
  ```

**Comment by Conaclos** [comment]
[Helix](https://github.com/helix-editor/helix)

**Reply by ematipico** [reply]
We already have a manual: https://github.com/rome/tools/blob/main/editors/helix/manual.md

EDIT: we'll have to update the documentation though, we have `rome lsp-proxy` command now, very useful for editors

**Reply by beeb** [reply]
Any update on that `lsp-proxy` command? Is it documented somewhere?

**Reply by ematipico** [reply]
> Any update on that `lsp-proxy` command? Is it documented somewhere?

It is https://docs.rome.tools/cli/#rome-lsp-proxy

**Reply by beeb** [reply]
Thanks @ematipico , I don't see mention of helix or helix configuration in that link. How is the configuration at https://github.com/rome/tools/blob/main/editors/helix/manual.md affected by this change?

**Reply by ematipico** [reply]
The manual was written by a third party contributor. The current core contributors don't know helix very much.

There's an issue you could follow that has some insights: https://github.com/rome/tools/issues/4107

It would be great if a community member could contribute to that

**Comment by andres-gr** [comment]
[Neovim](https://github.com/neovim/neovim)

**Reply by ematipico** [reply]
Are there any resources about how to create a neovim extension via LSP?

**Reply by IWANABETHATGUY** [reply]
For whom are interested in neovim configuration, here is a temporary config you could refer to https://github.com/rome/tools/discussions/2131#discussioncomment-2750526

**Reply by xanderjakeq** [reply]
i was able to use it with [formatter.nvim](https://github.com/mhartington/formatter.nvim):
```lua
filetype = {
    typescript = {
            function()
                return {
                    exe = "rome",
                    args = {"format", "--stdin-file-path", vim.fn.fnameescape(vim.api.nvim_buf_get_name(0))},
                    stdin = true
                }
            end
        },
}
```

**Reply by jellydn** [reply]
Works with [jose-elias-alvarez/null-ls.nvim: Use Neovim as a language server to inject LSP diagnostics, code actions, and more via Lua.](https://github.com/jose-elias-alvarez/null-ls.nvim) for formatting but does not support code_actions and diagnostic yet.

<img width="1338" alt="image" src="https://user-images.githubusercontent.com/870029/207780371-36c953d4-b745-4362-9e6f-7ff7a80825cb.png">

Read more on this discussion: https://github.com/jose-elias-alvarez/null-ls.nvim/discussions/1284#discussioncomment-4340235

**Comment by benjamineskola** [comment]
I've been trying to write an integration for Vim for my own use. It seems to me that integration with Vim/Neovim, from Rome's perspective, isn't so much a case of providing a plugin directly but about providing output in a format that other plugins can ingest. There are many different linter plugins (I've used at least three myself over the last few years) and it wouldn't be reasonable to expect Rome to write code for all of these, but rather to ensure that invoking Rome and ingesting its output is as simple as possible.

e.g., ~~ensuring that ANSI colour codes are not used when output is not to a terminal (i.e., applying `NO_COLOR` automatically)~~ (looks like my error here, sorry!) and providing a single-line output format for easy parsing.

I wouldn't be surprised if this is also true of Emacs.

**Reply by MichaReiser** [reply]
> e.g., ensuring that ANSI colour codes are not used when output is not to a terminal (i.e., applying NO_COLOR automatically) and providing a single-line output format for easy parsing.

This should already be the case today, andalso  the CLI supports a --no-colption too. 

> I've been trying to write an integration for Vim for my own use. It seems to me that integration with Vim/Neovim, from Rome's perspective, isn't so much a case of providing a plugin directly but about providing output in a format that other plugins can ingest

While this is great, it has the downside that users will need to know which plugins to install to get all the functionality Rome provides. I don't see this as a big issue in the short term, but our goal for the long term is to make it as easy as possible for users to set up rome. 

Nevertheless, interoperability is an important topic. Please open a discussion to discuss the specific additions you would need to Rome to make it work with the plugins.

**Reply by jeysal** [reply]
At least for nvim, the trend appears to be towards its native LSP integration as opposed to community plugins. This will mean both little effort for users to set Rome up and little effort for Rome to build custom things or document tens of alternatives for nvim.

While I'm still personally on `coc.nvim`, nvim native LSP is progressing fast, some people have already migrated to it, and I would expect it to become easier to configure too. I would not consider it worth it to start building or documenting any vim integrations other than a recommended setup with the native LSP client at this point.

**Reply by lwileczek** [reply]
I'm not sure if it would make it easier or more difficult but adding Rome as a linter/fixer in [ALE](https://github.com/dense-analysis/ale) would be great for vim and nvim

**Comment by philipheinser** [comment]
https://nova.app

**Comment by glerroo** [comment]
Gnome Builder

**Reply by reesericci** [reply]
https://wiki.gnome.org/Apps/Builder

**Comment by HolgerJeromin** [comment]
Visual Studio (non code)

**Comment by denbezrukov** [comment]
There is a workaround for JetBrains IDE to use Rome formatter.
https://youtrack.jetbrains.com/issue/WEB-46895/Support-for-Rome.js#focus=Comments-27-7386920.0-0
